src/Input.cpp

```diff
@@ -846,6 +857,18 @@ void Manager::DispatchEvents()
 
 */
 
+// Mouse rotation is enabled if:
+// MMB is pressed, or Ctrl+LMB is pressed, AND no other mouse buttons are pressed
+bool Manager::IsMouseRotatePressed()
```

Comment:
I am hesitant to accept this. I do not think (philosophically or practically) that it is the role of the `Input::Manager` to define the conditions for rotating the view across the entire application. I see `Input::Manager` as being *significantly* lower-level and this being the responsibility of a higher layer of the application.

Additionally, note that implementing it at this level provides no ability for the user to rebind which modifier key is used when emulating the middle mouse button (perhaps they'd like to use Alt or the meta keys?)

I wish I had an easy solution I could suggest to accomplish this. The mouse rotation code is one of those things that is common to a number of views, but specific enough in scope it's not really a good fit to throw in `Pi`, `Input::Manager`, or `View`. It's game-specific code, but there's a little bit of overlap with the editor as the editor also uses "game" widgets.

Reply:
> I am hesitant to accept this. I do not think (philosophically or practically) that it is the role of the `Input::Manager` to define the conditions for rotating the view across the entire application. I see `Input::Manager` as being _significantly_ lower-level and this being the responsibility of a higher layer of the application.
> 
Ok, just seemed like a natural/easy place for this functionality. It can be placed somewhere else instead. I'm very loathe to re-implement it in every place where it's needed though. I very much dislike code duplication and it already chafes me that I have to duplicate between these views and the Ship Spinner..

> Additionally, note that implementing it at this level provides no ability for the user to rebind which modifier key is used when emulating the middle mouse button (perhaps they'd like to use Alt or the meta keys?)
> 
Why not? I haven't gotten around to adding options for rebinding yet but I wasn't aware of there being a technical issue with adding such.

Reply:
> I'm very loathe to re-implement it in every place where it's needed though.

Same. I do think it should be centralized somewhere (perhaps a static method on some sort of helper namespace?) but just not in the low-level input subsystem.

> Why not? I haven't gotten around to adding options for rebinding yet but I wasn't aware of there being a technical issue with adding such.

The "no ability" referred to the as-implemented state. In terms of potential implementations, it's not so much that there is a technical problem with it, but rather that the API becomes significantly messier as the `Input::Manager` now has to export a specific action binding that all code which wants to use the function needs to ensure is present in the InputFrame stack before calling the method.